org.eclipse.dartboard/src/org/eclipse/dartboard/pub/PubGetHandler.java

```diff
@@ -0,0 +1,36 @@
+package org.eclipse.dartboard.pub;
+
+import org.eclipse.core.commands.AbstractHandler;
+import org.eclipse.core.commands.ExecutionEvent;
+import org.eclipse.core.commands.ExecutionException;
+import org.eclipse.core.resources.IResource;
+import org.eclipse.core.runtime.Adapters;
+import org.eclipse.core.runtime.preferences.InstanceScope;
+import org.eclipse.dartboard.Constants;
+import org.eclipse.jface.viewers.ISelection;
+import org.eclipse.jface.viewers.IStructuredSelection;
+import org.eclipse.ui.handlers.HandlerUtil;
+import org.eclipse.ui.preferences.ScopedPreferenceStore;
+
+public class PubGetHandler extends AbstractHandler {
+
+	private PubService pub;
+
+	private ScopedPreferenceStore preferences;
+
+	public PubGetHandler() {
+		pub = PubService.getInstance();
```

Comment:
Same here also. You can directly call the static method instead of creating an object.

Reply:
PubService is supposed to be a Singleton. And since it has some Data stored, it would not be ideal to make it all static.

Ideally I would like to provide it as an OSGi service and have it injected into the classes I need it. This however is not possible for pages (see DartPreferencePage) that are defined via extension points. At least I could not figure out how to do so.

Once we have a working OSGi service structure for this, the refactoring would be minimal, as we can just add a `@Inject` to the field instead of changing all method calls to be non-static. That's why I used the Singlet pattern for now.